src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -510,7 +510,7 @@ def __setattr__(self, name: str, value: Any):
         if hasattr(self, name) and hasattr(self.config, name):
             # We need to overwrite the config if name exists in config
             if isinstance(getattr(self.config, name), (tuple, list)):
-                if self.config[name][0] is not None:
+                if value is not None and self.config[name][0] is not None:
                     class_library_tuple = (value.__module__.split(".")[0], value.__class__.__name__)
                 else:
```

Comment:
the setting value can be None with a defined class library tuple on the config -- i.e. if the text encoder is an optional  component